File: 1448-maximum-69-number/maximum-69-number.java
class Solution {
    public int maximum69Number (int num) {
        if (num == 9999) {
            return num;
        }
        if (num == 96 || num == 69) {
            return 99;
        }
        if (num == 996) {
            return 999;
        }

        String a = String.valueOf(num);
        char[] l = a.toCharArray();

        if (num == 966 || num == 969) {
            l[1] = '9';
            int n = Integer.parseInt(new String(l));
            return n;
        }
        if (num <= 6999) {
            l[0] = '9';
            int n = Integer.parseInt(new String(l));
            return n;
        }
        if (num >= 9666 && num <= 9699) {
            l[1] = '9';
            int n = Integer.parseInt(new String(l));
            return n;
        }
        if (num == 9966 || num == 9969) {
            l[2] = '9';
            int n = Integer.parseInt(new String(l));
            return n;
        }
        if (num == 9996) {
            l[3] = '9';
            int n = Integer.parseInt(new String(l));
            return n;
        }

        return num;
    }
}

File: 1448-maximum-69-number/maximum-69-number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        if num == 9999:
            return num
        if num == 96 or num == 69:
            return 99
        if num == 996:
            return 999
        a = str(num)
        l = list(a)
        if num == 966 or num == 969:
            l[1] = '9'
            s = int(''.join(map(str,l)))
            return s
        if num >= 9666 and num <= 9699:
            l[1] = '9'
            s = int(''.join(map(str,l)))
            return s
        if num == 9966 or num == 9969:
            l[2] = '9'
            s = int(''.join(map(str,l)))
            return s
        if num == 9996:
            l[3] = '9'
            s = int(''.join(map(str,l)))
            return s
        if num <= 6999:
            l[0] = '9'
            s = int(''.join(map(str,l)))
            return s
        
            